diff --git a/etc/modulesets/jhb/gtk-osx-bootstrap.modules b/etc/modulesets/jhb/gtk-osx-bootstrap.modules
index 140a570..bd57923 100644
--- a/etc/modulesets/jhb/gtk-osx-bootstrap.modules
+++ b/etc/modulesets/jhb/gtk-osx-bootstrap.modules
@@ -24,7 +24,7 @@
               href="https://github.com/"
               type="tarball" />
   <repository name="nasm"
-              href="https://www.nasm.us/pub/nasm/releasebuilds/"
+              href="https://mirror.dotsrc.org/mirrors/mirrors/pub/kde-applicationdata/craft/3rdparty/nasm/"
               type="tarball" />
   <!---->
   <autotools id="libpng"
